http/dotnet/httpclient/src/HttpCore.cs

```diff
@@ -182,7 +165,7 @@ private async Task<HttpResponseMessage> GetHttpResponseMessage(RequestInfo reque
             if(requestInfo == null)
                 throw new ArgumentNullException(nameof(requestInfo));
 
-            await AddBearerIfNotPresent(requestInfo);
+            await authProvider.AuthenticateRequest(requestInfo);
```

Comment:
I just realized that a user now must define an `authProvider` to use the sdk. Even if they choose to go a different way to authenticate like adding a custom delegating handler in the pipeline of their custom httpClient. 
Is it worth considering allowing an null `authProvider` ?

Reply:
This is something I debated myself over for a while and still don't have a strong opinion about.
On one hand, people might now want any auth, why force them to have a provider?
On the other hand, you want the API surface to be as explicit as possible and lead people to do the right thing. By requiring an auth provider in the constructor of the service, we're "nudging them" to recognize it's needed and to make a choice about that. (intentional/semantic programming).
I don't believe going in with a middleware approach is the best thing to do because it ties the authentication to a specific http implementation, and one of the promises of Kiota was to provide separation of concerns on those aspect (+ serialization).